src/iceberg/util/checked_cast.h

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#pragma once
```

Comment:
nit: add the `\file` docstring and mention that it is adapted from Apache Arrow.

Reply:
Will add later today, thanks for the suggestion.

Reply:
If we take big pieces from other projects, then we also want to add that to the `LICENSE`:

For example in PyIceberg: https://github.com/apache/iceberg-python/blob/260ef54e3920d435ae3b2ccda090e66f9c1ac015/LICENSE#L206-L213